packages/vue/src/errorhandler.ts

```diff
@@ -17,10 +17,10 @@ export const attachErrorHandler = (app: Vue, options: Options): void => {
       trace,
     };
 
-    if (vm && options.attachProps) {
+    if (options.attachProps && vm && (vm.$options?.propsData || vm.$props)) {
       // Vue2 - $options.propsData
       // Vue3 - $props
-      metadata.propsData = vm.$options.propsData || vm.$props;
+      metadata.propsData = vm.$options?.propsData || vm.$props;
```

Comment:
> Optional chains are transpiled to especially large code

I had no idea 😅 

> What do you think?

I'll have to suggest something else as it breaks some tests I wrote
![Screenshot 2022-06-20 at 12 06 00](https://user-images.githubusercontent.com/10957531/174579193-f5ae6540-7c19-4676-ad97-d086ec117f07.png)

Especially because in your suggestion, the outer condition prevents the attachement of props for vue 3 (`$props` will be defined, but won't satisfy the condition)

This one, less compact, works:
```ts
    if (options.attachProps && vm) {
      // Vue2 - $options.propsData
      // Vue3 - $props
      if (vm.$options && vm.$options.propsData) {
        metadata.propsData = vm.$options.propsData;
      } else if (vm.$props) {
        metadata.propsData = vm.$props;
      }
    }
```

Would it work for you ?

Reply:
Ah yeah you're totally right - my bad. The last snippet you shared works. Thank you!